### DISCUSSION TITLE
Empty Project with Refactored + XT fails to build

### DISCUSSION THREAD

**Discussion by mmodi3** [Voidware-Prohibited/ALSXT#65] [Q&A] [answered]
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2143	syntax error: missing ';' before '*'	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	45	
Error	C4430	missing type specifier - int assumed. Note: C++ does not support default-int	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	45	
Error	C2334	unexpected token(s) preceding '{'; skipping apparent function body	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	45	
Error	C2065	'UAlsCameraComponent': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	48	
Error	C2923	'TObjectPtr': 'UAlsCameraComponent' is not a valid template type argument for parameter 'T'	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	48	
Error	C2955	'TObjectPtr': use of class template requires template argument list	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	48	
Error	C2039	'Character': is not a member of 'UALSXTIdleAnimationComponent'	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Intermediate\Build\Win64\UnrealEditor\Inc\ALSXT\UHT\ALSXTIdleAnimationComponent.gen.cpp	647	
Error	C2618	illegal member designator in offsetof	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Intermediate\Build\Win64\UnrealEditor\Inc\ALSXT\UHT\ALSXTIdleAnimationComponent.gen.cpp	647	
Error	C2512	'TObjectPtr': no appropriate default constructor available	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Intermediate\Build\Win64\UnrealEditor\Inc\ALSXT\UHT\ALSXTIdleAnimationComponent.gen.cpp	773	
Error	C2143	syntax error: missing ';' before '*'	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	45	
Error	C4430	missing type specifier - int assumed. Note: C++ does not support default-int	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	45	
Error	C2334	unexpected token(s) preceding '{'; skipping apparent function body	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	45	
Error	C2065	'UAlsCameraComponent': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	48	
Error	C2923	'TObjectPtr': 'UAlsCameraComponent' is not a valid template type argument for parameter 'T'	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	48	
Error	C2955	'TObjectPtr': use of class template requires template argument list	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Public\Components\Character\ALSXTIdleAnimationComponent.h	48	
Error	C2512	'TObjectPtr': no appropriate default constructor available	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	9	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	37	
Error	C2065	'AALSXTCharacter': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	37	
Error	C2672	'Cast': no matching overloaded function found	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	37	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	39	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	41	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	51	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	75	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	75	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	76	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	193	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	234	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	282	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	282	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	282	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	282	
Error	C2660	'UALSXTIdleAnimationComponent::SelectIdleAnimations': function does not take 2 arguments	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	282	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	287	
Error	C2065	'Character': undeclared identifier	ALSBasedFPController	C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\Plugins\ALSXT-master\Source\ALSXT\Private\Components\Character\ALSXTIdleAnimationComponent.cpp	299	
Error	MSB3073	The command ""C:\Program Files\Epic Games\UE_5.3\Engine\Build\BatchFiles\Build.bat" ALSBasedFPControllerEditor Win64 Development -Project="C:\Users\modim\Documents\Unreal Projects\ALSBasedFPController\ALSBasedFPController.uproject" -WaitMutex -FromMsBuild" exited with code 6.	ALSBasedFPController	C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.MakeFile.Targets	44

**Comment by colorindarkness** [comment]
Thank you I think I forgot to include the ALSXTCharacter Header file. I made this an issue [here](https://github.com/Voidware-Prohibited/ALSXT/issues/66) and am currently fixing it

**Comment by colorindarkness** [comment, marked as answer]
Should now be fixed in [e0913af](https://github.com/Voidware-Prohibited/ALSXT/commit/e0913af86e75e2ff682c5f4ab42ed60300d8df24)
